build-logic/settings.gradle.kts

```diff
@@ -5,3 +5,7 @@ dependencyResolutionManagement {
         }
     }
 }
+
+plugins {
+    id("org.gradle.toolchains.foojay-resolver-convention") version "0.5.0"
```

Comment:
This is the mechanism that downloads missing java distributions - https://docs.gradle.org/8.2/userguide/toolchain_plugins.html
If these were missing, the build would fail if the runner didn't have all required java versions available locally

Reply:
The plugin has to be applied to both projects ("project" represented by single `settings.gradle` file), since each of them has their own toolchain config